eq-author/src/components/RichTextEditor/PipingMenu.js

```diff
@@ -126,6 +126,44 @@ const calculateEntityName = ({ pageId, confirmationId }) => {
   return "section";
 };
 
+const postProcessPipingContent = ( entity ) => {
+  if (!entity) {
+    return false;
+  }
+
+  const processedAnswers = [];
+
+  entity.availablePipingAnswers.forEach ( answer => {
+    
+    
+    if (answer.type === "DateRange") {
+      processedAnswers.push(
+        {
+          ...answer,
+          id: `${answer.id}from`,
+        }
+      );
+
+      processedAnswers.push(
+        {
+          ...answer,
+          id: `${answer.id}to`,
+          displayName: answer.secondaryLabel
+        }
+      );
+
+    } else {
+      processedAnswers.push(answer);
+    }
+   
+  });
+
+  return {
+    ...entity,
+    availablePipingAnswers: processedAnswers
```

Comment:
Just wondering about having this logic live in the front end as it feels a little bit like it should be a concern of the api to return the availablePipingAnswers correctly rather than forcing the f/e to apply additional logic to the query result?

Reply:
I was thinking the same thing. The only weird thing is that `DateRange` is an answer type and not a composite answer as it was before.

If this is only returning answers then it would seem odd if it would expand just DateRange out. 

So I see we have two options if we do want to push it the API:
1. Bring back composite answer and have this return the 2 Date answers from it.
2. Accept the complexity of having DateRange be special

Reply:
Ideally I think it would be nice to keep that composite answer as it seems to be closer to the actual model of a DateRange answer. However for now maybe it's worth just doing option 2, use the logic that's already inside this Pr and just push it into the API?

Reply:
Yeah, discussed with @samiwel and @pricea1 as we only have one instance of this complex answer we will leave it as is for the moment.

Reply:
After further investigation, adding to eq-author-api proved to create more issues. Changing the ID at this level lost the answers connection to the page and section, which used the answer ID as the connection. After discussion with @samiwel and @tgandrews it was decided to use the first approach and keep in frontend and tackle the issue as a bigger task at a later date.